### ISSUE TITLE
Uncaught TypeError: (var)[Symbol.iterator] is not a function

### ISSUE THREAD

**Issue by dcro** [atom/find-and-replace#851]
[Enter steps to reproduce:]

1. Enter `show(` in the search field and select a regex search
2. Open any new editor and type some text (or remove some text)

![screenshot 2017-02-07 17 32 16](https://cloud.githubusercontent.com/assets/1079773/22697912/9a590940-ed5b-11e6-9acc-8346a2c92907.png)

**Atom**: 1.14.0-beta2 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.12.2
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package 0.205.1

### Stack Trace

Uncaught TypeError: (var)[Symbol.iterator] is not a function

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/buffer-search.js:227

TypeError: (var)[Symbol.iterator] is not a function
    at BufferSearch.bufferStoppedChanging (/app.asar/node_modules/find-and-replace/lib/buffer-search.js:227:93)
    at Function.module.exports.Emitter.simpleDispatch (/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at /app.asar/node_modules/text-buffer/lib/text-buffer.js:1368:25
```

### Commands

```
  2x -6:31.7.0 editor:move-to-first-character-of-line (input.hidden-input)
     -6:31.5.0 core:save (input.hidden-input)
     -6:28.3.0 core:move-down (input.hidden-input)
  2x -6:28 editor:move-to-first-character-of-line (input.hidden-input)
     -6:27.6.0 core:select-down (input.hidden-input)
     -6:27.2.0 core:backspace (input.hidden-input)
     -6:26.5.0 core:save (input.hidden-input)
     -6:25.9.0 editor:move-to-end-of-screen-line (input.hidden-input)
     -6:25.7.0 editor:move-to-first-character-of-line (input.hidden-input)
     -6:25.2.0 editor:select-to-first-character-of-line (input.hidden-input)
  2x -6:24.9.0 core:backspace (input.hidden-input)
     -6:24.4.0 core:save (input.hidden-input)
     -6:24 editor:newline (input.hidden-input)
     -6:22.6.0 core:save (input.hidden-input)
  2x -6:22.3.0 editor:move-to-first-character-of-line (input.hidden-input)
     -0:02.1.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
alignment 1.1.5 
api-docs 0.0.6 
atom-autocomplete-php 0.21.0 
atom-beautify 0.29.17 
atom-typescript 10.1.13 
base64 0.3.0 
blade-snippets 0.2.0 
editorconfig 2.2.0 
elasticsearch-client 0.9.5 
file-icons 2.0.13 
language-blade 0.27.1 
language-docker 1.1.7 
language-dots 0.0.4 
language-nginx 0.6.2 
laravel 0.7.1 
linter 1.11.21 
linter-eslint 8.1.0 
linter-js-yaml 1.2.6 
linter-php 1.3.1 
linter-phpcs 1.5.8 
linter-stylelint 3.7.0 
merge-conflicts 1.4.4 
pigments 0.39.0 
pretty-json 1.6.1 
project-manager 3.3.3 
```

**Comment by 50Wliu** [comment]
Yikes!  Thanks for the bug report - I can confirm on Atom 1.15.0-dev-4eb3f22, Windows 10 x64.

**Comment by adambutler** [comment]
Same issue here:

**Atom**: 1.14.2 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.12.3
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package 0.206.1


### Stack Trace

Uncaught TypeError: (var)[Symbol.iterator] is not a function

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/buffer-search.js:227

TypeError: (var)[Symbol.iterator] is not a function
  at BufferSearch.bufferStoppedChanging (/app.asar/node_modules/find-and-replace/lib/buffer-search.js:227:93)
  at Function.module.exports.Emitter.simpleDispatch (/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
  at Emitter.module.exports.Emitter.emit (/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
  at /app.asar/node_modules/text-buffer/lib/text-buffer.js:1368:25

```

### Commands

```
  2x -0:57.8.0 core:move-left (input.hidden-input)
 11x -0:56.7.0 core:select-right (input.hidden-input)
     -0:55.8.0 core:select-left (input.hidden-input)
     -0:55.5.0 core:move-left (input.hidden-input)
  9x -0:54.8.0 core:move-right (input.hidden-input)
     -0:53.8.0 core:select-left (input.hidden-input)
     -0:53.4.0 editor:select-to-first-character-of-line (input.hidden-input)
 26x -0:52.9.0 find-and-replace:select-next (input.hidden-input)
  2x -0:50.1.0 editor:consolidate-selections (input.hidden-input)
     -0:47 core:cancel (input.hidden-input)
     -0:45.7.0 editor:select-to-first-character-of-line (input.hidden-input)
     -0:45.3.0 core:backspace (input.hidden-input)
     -0:44.9.0 core:move-up (input.hidden-input)
     -0:44.4.0 core:move-down (input.hidden-input)
     -0:44 core:backspace (input.hidden-input)
     -0:43.3.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-alignment 0.12.1 
emmet 2.4.3 
file-icons 2.0.15 
git-plus 7.3.2 
language-html-swig 2.0.1 
linter-rubocop 0.5.3 
minimap 4.26.8 
minimap-cursorline 0.2.0 
nuclide 0.207.0 
pigments 0.39.0 
slugify-atom 0.3.0 
Sublime-Style-Column-Selection 1.7.3 
```

**Comment by 50Wliu** [comment]
Fixed by #860.  @adambutler please update to 1.14.3.
